### PR TITLE
Graph Table Totals Row and No Value Group targets

### PR DESCRIPTION
**Pull request by Elkrival**
### Graph Table Totals Row 
Display percentages only if all sites have a target for a value
link to chat discussion https://gnardog.slack.com/archives/C03KGHSSTNC/p1666195037476419
Screen Shot 2022-10-21 at 11 59 02 AM
Screen Shot 2022-10-21 at 11 59 12 AM
Screen Shot 2022-10-21 at 11 59 27 AM

### Skip No Value Groups when calculating sites targetTotal
Screen Shot 2022-10-21 at 12 36 30 PM

### CODE REVIEW

File: server/controllers/chartsController.js [hunk 1]
```diff
@@ -249,3 +244,13 @@ function isAnyTargetIncluded(studyTotals) {
     .filter((site) => site !== TOTALS_STUDY)
     .some((site) => studyTotals[site]?.targetTotal !== undefined)
 }
+
+function isTargetValueOnAllSites(fieldLabelValueMap, valueLabel) {
+  return Object.values(
+    fieldLabelValueMap.find(({ label }) => label === valueLabel).targetValues
+  ).every((target) => !!target)
```
Comment (mikestone14) [inline server/controllers/chartsController.js]: Minor suggestion to make this more readable.

```suggestion
  const fieldTargetValues = fieldLabelValueMap.find(({ label }) => label === valueLabel).targetValues
  
  return Object.values(fieldTargetValues).every(Boolean)
```

File: server/controllers/chartsController.js [hunk 2]
```diff
@@ -249,3 +244,13 @@ function isAnyTargetIncluded(studyTotals) {
...
+}
+
+function isNoValueGroup(value) {
+  return value === ''
```
Comment (mikestone14) [inline server/controllers/chartsController.js]: can we extract the hard-coded `''` to a constant to make the code more self-documenting?

File: server/controllers/chartsController.js [hunk 3]
```diff
@@ -24,22 +24,17 @@ const postProcessData = (data, studyTotals) => {
     const percent = studyCountsToPercentage(count, totals)
     const existingEntriesForStudy = processedDataBySite.get(study)
     const hasTargetValue = !!targetValue && study !== TOTALS_STUDY
-    const isTargetValueMissing = !targetValue && study !== TOTALS_STUDY
 
-    if (hasTargetValue) {
+    if (
+      hasTargetValue &&
+      isTargetValueOnAllSites(chart.fieldLabelValueMap, valueLabel)
```
Comment (mikestone14) [inline server/controllers/chartsController.js]:
```suggestion
      hasTargetValueOnAllSites(chart.fieldLabelValueMap, valueLabel)
```

Comment (Elkrival) [reply server/controllers/chartsController.js]: [cbdb19e](https://github.com/AMP-SCZ/dpdash/pull/217/commits/cbdb19ee4098d9b8d76793cb0d383fe9ad4e86d7)

File: server/controllers/chartsController.js (hunk 2 above)
Comment (Elkrival) [reply server/controllers/chartsController.js]: [cbdb19e](https://github.com/AMP-SCZ/dpdash/pull/217/commits/cbdb19ee4098d9b8d76793cb0d383fe9ad4e86d7)

File: server/controllers/chartsController.js (hunk 1 above)
Comment (Elkrival) [reply server/controllers/chartsController.js]: [cbdb19e](https://github.com/AMP-SCZ/dpdash/pull/217/commits/cbdb19ee4098d9b8d76793cb0d383fe9ad4e86d7)

**Pull request closed without merging**